core/include/kdl_codyco/sensors.hpp

```diff
@@ -0,0 +1,213 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details
+ */
+
+#ifndef IDYNTREE_CORE_SENSORS_HPP
+#define IDYNTREE_CORE_SENSORS_HPP
+
+namespace KDL {
+    class Wrench;
+}
+
+
+#include <string>
+
+namespace KDL {
+namespace CoDyCo {
+
+    enum SensorType
+    {
+        SIX_AXIS_FORCE_TORQUE = 0,
+        NR_OF_SENSOR_TYPES = 1
+    };
+
+     /**
+     * Virtual interface to Sensor Class.
+     * All sensor class inherit from this base class.
+     *
+     */
+    class Sensor {
+
+    public:
+        /**
+         * Get the id (name) of sensor.
+         */
+        virtual std::string getName() const = 0;
+
+        /**
+         * Get the type of the sensor.
+         * @return the type of the sensor.
+         */
+        virtual SensorType getSensorType() const = 0;
+
+        /**
+         * Get the id (name) of the parent entity (Junction or Link).
+         */
+        virtual std::string getParent() const = 0;
+
+        /**
+         * Get the numeric index of the parent of the sensor.
+         * Depending on the type of the sensor, the parent could be
+         * a Junction or a Link.
+         *
+         * @return the index of the parent (Junction or Link) of the sensor.
+         */
+        virtual int getParentIndex() const = 0;
+
+        /**
+         * Return true if the sensor has been appropriately configured (all
+         * setters where setted, false otherwise.
+         *
+         */
+        virtual bool isValid() const = 0;
+    };
+
+    /**
+     * Structure representing a group of sensors associated with an UndirectedTree.
+     */
+    class SensorsTree {
+            struct SensorsTreePimpl;
+            SensorsTreePimpl * pimpl;
+        public:
+
+            /**
+             * Constructor.
+             */
+            SensorsTree();
+
+            /**
+             * Copy constructor
+             */
+            SensorsTree(const SensorsTree& other);
+
+            /**
+             * Copy operator
+             */
+            SensorsTree& operator=(const SensorsTree &other);
+
+            /**
+             * Destructor.
+             */
+            virtual ~SensorsTree();
+
+            /**
+             * Add a sensor to the SensorsTree.
+             * The sensor index will depend on the order in which the
+             * sensor are added to the sensorsTree.
+             *
+             * The passed sensor will be dynamic casted to the specified sensor type,
+             * and will be copied in the sensors tree only if the dynamic cast will be successful.
+             *
+             * @param[in] sensor_type the SensorType of the added sensor.
+             * @param[in] a constant reference to the Sensor to add.
+             * @return the sensor index of the newly added sensor, or -1 in case of error.
+             */
+            int addSensor(const SensorType & sensor_type, const Sensor & sensor);
```

Comment:
Done in https://github.com/robotology-playground/idyntree/commit/cbba345b64ba70fd7b9a45d9a96911da51ce58e5